tests/load/pipeline/test_snowflake_pipeline.py

```diff
@@ -204,3 +206,66 @@ def test_char_replacement_cs_naming_convention(
     results = rel_.fetchall()
     assert len(results) == 1
     assert "AmlSistUtfoertDato" in rel_.columns_schema
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        all_staging_configs=True,
+        with_file_format="parquet",
+        subset=["snowflake"],
+    ),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize(
+    "use_vectorized_scanner",
+    ["TRUE", "FALSE"],
+)
+def test_snowflake_use_vectorized_scanner(destination_config, use_vectorized_scanner: str):
+    """Tests whether the vectorized scanner option is correctly applied when loading Parquet files into Snowflake."""
+
+    def get_copy_into_queries(client: SqlClientBase[Any], table_name: str):
+        """Fetch COPY INTO queries within the last minute containing the table name."""
+        query = """
+        SELECT QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY( \
+            END_TIME_RANGE_START => DATEADD(MINUTE, -5, CURRENT_TIMESTAMP), \
+            END_TIME_RANGE_END => CURRENT_TIMESTAMP \
+        )) WHERE QUERY_TEXT LIKE '%COPY INTO%'
+        """
+        return [row[0] for row in client.execute_sql(query) if table_name in row[0]]
+
+    os.environ["DESTINATION__SNOWFLAKE__USE_VECTORIZED_SCANNER"] = use_vectorized_scanner
+
+    pipeline, data = simple_nested_pipeline(
+        destination_config, f"vectorized_scanner_{use_vectorized_scanner}_{uniq_id()}", False
+    )
+    info = pipeline.run(data(), **destination_config.run_kwargs)
+    assert_load_info(info)
+
+    # Allow time for query history to update in Snowflake
+    time.sleep(2)
```

Comment:
this is a good way to check if a proper query was generated. but there are other ways that IMO require way less work and are pretty standard
* refactor the code so you can unit test the essential function. this is already done with `gen_copy_sql`
* you can also using `Mocker` and `spy` to hijack the return value if the function value and check if VECTORIZE part is present
then you do not need to check Snowflake history - you can trust that the other tests are checking if the `gen_copy_sql` output is really executed

tldr;> nothing to change. but you can read more on mocking